runtime/caml/domain.h

```diff
@@ -0,0 +1,37 @@
+/**************************************************************************/
+/*                                                                        */
+/*                                 OCaml                                  */
+/*                                                                        */
+/*          Xavier Leroy and Damien Doligez, INRIA Rocquencourt           */
+/*                                                                        */
+/*   Copyright 1996 Institut National de Recherche en Informatique et     */
+/*     en Automatique.                                                    */
+/*                                                                        */
+/*   All rights reserved.  This file is distributed under the terms of    */
+/*   the GNU Lesser General Public License version 2.1, with the          */
+/*   special exception on linking described in the file LICENSE.          */
+/*                                                                        */
+/**************************************************************************/
+
+#ifndef CAML_DOMAIN_H
+#define CAML_DOMAIN_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifdef CAML_INTERNALS
+
+#include "domain_state.h"
+
+void caml_init_domain(void);
+
+#endif /* CAML_INTERNALS */
+
+CAMLextern caml_domain_state* Caml_state;
```

Comment:
This I think will result in an error if `CAML_INTERNALS` isn't defined - should it be within the block too?

Reply:
Removed this extern declaration as it was superfluous; `domain_state.h` declares the extern which is included in this file.